### PR TITLE
Issue37 foldermapping v1

### LINKED ISSUE #37: Category pages with article list
Basically a list of articles for a category. Uses `article-list` block with a different filter. 

See `fstab.yaml` how folders are mapped to  `/category/default` and `/tag/default`. 

example:
https://www.24life.com/category/lifestyle/

### PR DESCRIPTION
**Pull request by meejain**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #37 

Test URLs:
- Before: https://main--24life--hlxsites.hlx.page/category/lifestyle
- After: https://issue37-foldermapping-v1--24life--hlxsites.hlx.page/category/lifestyle

### CODE REVIEW

File: fstab.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ mountpoints:
   /: https://adobe.sharepoint.com/sites/HelixProjects/Shared%20Documents/sites/24life
 
 folders:
-  /category/: /category/default
+  /category/: /category/default
```
Comment (iamitshri) [inline fstab.yaml:5]: only a new line was added.

File: blocks/article-list/article-list.js [hunk 2]
```diff
@@ -8,7 +8,16 @@ import { createCardBlock } from '../card/card.js';
...
   const filters = removeEmptyKeyOrValue(readBlockConfig(block));
   const isEmptyFilter = Object.keys(filters).length === 0;
-  if (isEmptyFilter && document.location.pathname.startsWith('/author/')) {
+  if (isEmptyFilter && document.location.pathname.startsWith('/category/')) {
+    // auto-detect category, e.g. https://www.24life.com/category/lifestyle
+    const parts = document.location.pathname.split('/');
+    const askedCategory = parts[parts.length - 1];
+    filters.categories = askedCategory.replace(/-/g, ' ');
+    const firstContainer = document.createElement('div');
+    // firstContainer.innerHTML = `<h1>${askedCategory.toUpperCase()}</h1>`;
```
Comment (iamitshri) [inline blocks/article-list/article-list.js:17]: comment can be removed :)

File: blocks/article-list/article-list.js [hunk 3]
```diff
@@ -8,7 +8,15 @@ import { createCardBlock } from '../card/card.js';
...
   const filters = removeEmptyKeyOrValue(readBlockConfig(block));
   const isEmptyFilter = Object.keys(filters).length === 0;
-  if (isEmptyFilter && document.location.pathname.startsWith('/author/')) {
+  if (isEmptyFilter && document.location.pathname.startsWith('/category/')) {
+    // auto-detect category, e.g. https://www.24life.com/category/lifestyle
+    const parts = document.location.pathname.split('/');
+    const askedCategory = parts[parts.length - 1];
+    filters.categories = askedCategory.replace(/-/g, ' ');
+    const firstContainer = document.createElement('div');
+    firstContainer.innerHTML = `<h1>IN: ${filters.categories.toUpperCase()}</h1>`;
```
Comment (rrusher) [inline blocks/article-list/article-list.js:17]: This text is displaying twice.
The text reads "BY: IN: LIFESYTLE", it should be "IN: LIFESTYLE". It does this via the Author templates' author.css. Just remove the metadata block from default.docx.

**Review by rrusher — CHANGES_REQUESTED** [review]

File: blocks/article-list/article-list.js (hunk 3 above)
Comment (meejain) [reply blocks/article-list/article-list.js:17]: yes @rrusher  - I did remove the metadata block from default but still we are seeing that - https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/lifestyle . This issue is happening only on this page. Rest all the categories are popping up fine - 

https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Discover
https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Fashion
https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Flexibility
https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Game-Changers
https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Motivate
https://issue37-foldermapping-v1--24life--hlxsites.hlx.live/category/Motoring

Seems like some issue with the caching at the CDN, may be. Please advise.

File: blocks/article-list/article-list.js (hunk 2 above)
Comment (meejain) [reply blocks/article-list/article-list.js]: issue fixed

**Review by rrusher — APPROVED** [review]

**Pull request merged**